anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2604,3 +2621,18 @@ def upsert_eval(dbsession, userId, record):
 
 
 ################################################################################
+
+# return true or false if image is a valid size based upon max_compressed_image_size specified in config
+def is_image_valid_size(image_info):
+    localconfig = anchore_engine.configuration.localconfig.get_config()
+    max_compressed_image_size = localconfig.get("max_compressed_image_size")
```

Comment:
Would we ever want to add a maximum possible configured value here? IE. max_compressed_image_size can't go over 10GB or something? and it can't be negative?

If the configured value is out of expected bounds perhaps we should do some normalizing. @zhill wdyt?

Reply:
If the configured value is out of acceptable bounds then the system should fail bootstrap with an error indicating invalid configuration found. We could set a max allowed configurable size, but I don't see that as an error case to handle it's just effectively a no-op. A value < 0 should be considered a "disabled" state equivalent to the config value being omitted entirely.

Reply:
@zhill I added a validation that prevents the service from starting if the value is not an integer. The way the code is structured, a negative value provided should always cause the image to pass this validation because the size provided by the manifest should never be less than 0, but I went ahead and added an explicit check on the condition to verify that the config value is "enabled"